### ISSUE TITLE
get query parameters at any point

### ISSUE THREAD

**Issue by avdhootu27** [vaadin/flow#19064]
### Describe your motivation

Currently there is no API which can return query parameters of current page url, or there is also not any API which returns current url path.

### Describe the solution you'd like

Provide an API which returns at least query parameters, or whole url path.

**Comment by mcollovati** [comment]
Couldn't you use `UI.getActiveViewLocation()`?

**Comment by avdhootu27** [comment]
I am on Vaadin 14 and there is no getActiveViewLocation method in UI

**Comment by mshabarov** [comment]
Yeah, this API wasn't picked to Vaadin 14 from Vaadin 24, but you can try `UI.getInternals().getActiveViewLocation().getQueryParameters()`, which should be already in Vaadin 14 for years.

**Comment by avdhootu27** [comment]
I tried it but, it doesn't work properly, mostly it worked first time but later when I switched between views it didn't work.
